Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -1358,7 +1358,10 @@ public boolean cantBeAttachedBy(MageObject attachment, Ability source, Game game
                 && attachmentPermanent != null
                 && attachmentPermanent.getSpellAbility() != null
                 && !attachmentPermanent.getSpellAbility().getTargets().isEmpty()) {
-            canAttach = game.getPermanent(attachment.getId()).getSpellAbility().getTargets().get(0).getFilter().match(this, game);
+            Filter filter = attachmentPermanent.getSpellAbility().getTargets().get(0).getFilter();
+            if (filter instanceof FilterPermanent) {
+                canAttach = ((FilterPermanent) filter).match(this, attachmentPermanent.getControllerId(), source, game);
+            }
```

Comment:
No; in fact, doing so causes the tests with Spellweaver Volute (enchant card in graveyard) and Psychic Possession (enchant opponent) to now pass.

Reply:
Yeah I was expecting that, since they respectively have a `FilterCard` and a `FilterPlayer` for target.

The attach to permanent action should now work properly.
I guess if we ever have an attach/reattach not specific to permanents, there would need to be `cantBeAttachedBy` in `Card`/`Player` that would look for their own `Filter`.

If there is ever an Aura able to enchant both `Permanent` and `Player`, I guess the engine will burst into flames.